spec/features/user_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+RSpec.describe 'Users', type: :request do
+  describe 'GET /users/:id' do
+    let!(:user) { User.create(name: 'John Doe') }
+    let!(:post1) { Post.create(author: user, title: 'Post 1', text: 'This is post 1') }
+    let!(:post2) { Post.create(author: user, title: 'Post 2', text: 'This is post 2') }
+    let!(:post3) { Post.create(author: user, title: 'Post 3', text: 'This is post 3') }
+    before { get "/users/#{user.id}" }
+    it 'displays the user\'s username' do
+      expect(response.body).to include(user.name)
+    end
+    it 'displays the user\'s first 3 posts' do
+      expect(response.body).to include(post1.title)
+      expect(response.body).to include(post2.title)
+      expect(response.body).to include(post3.title)
+    end
+    it 'displays a button to view all user\'s posts' do
+      expect(response.body).to include(user_posts_path(user))
+    end
+  end
```

Comment:
- [ ] You're missing writing most of the tests as requested in the project requirements. Please make sure to write integration tests for all features as requested in the project [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=User%20index%20page,each%20commentor%20left.). Please also separate tests for index and show actions in different files for users and posts.